src/interfaces.d.ts

```diff
@@ -25,6 +25,12 @@ export interface ChildNodeFunction {
 	(this: Widget<WidgetState, WidgetProperties>): DNode[];
 }
 
+export interface PropertiesChangeEvent<T> extends EventTypedObject<'properties:changed'> {
```

Comment:
I think we should look to generify the `properties` attribute and add the `on(...` signature to the `WidgetOverrides` type, this is so that a user can then get the `on` intellisense for the `properties:changed` evnet.